core/src/test/java/com/google/cloud/sql/core/CoreSocketFactoryTest.java

```diff
@@ -145,15 +143,16 @@ public void setup() throws IOException, GeneralSecurityException, ExecutionExcep
 
     defaultExecutor = CoreSocketFactory.getDefaultExecutor();
 
-    // Mock the API client for testing
+    // Stub the API client for testing
     when(adminApi.instances()).thenReturn(adminApiInstances);
 
-    // Mock wrong project and instance
-    doThrow(fakeNotConfiguredException()).when(adminApiInstances).get(anyString(), anyString());
-    // Mock correct project, wrong instance
-    doThrow(fakeNotAuthorizedException()).when(adminApiInstances).get(eq("myProject"), anyString());
-    // Mock correct instance
-    doReturn(adminApiInstancesGet).when(adminApiInstances).get(eq("myProject"), eq("myInstance"));
+    // Stub when generic cases for project/instance
+    when(adminApiInstances.get(anyString(), anyString())).thenThrow(fakeNotConfiguredException());
+    // Stub when correct project, but generic instance
+    when(adminApiInstances.get(eq("myProject"), anyString()))
```

Comment:
Wait! I thought you said you had to use `doThrow()` here because otherwise this would actually throw because of the previous throw instruction. Does this work?

Reply:
Yes, this seems to be working fine. The actual issue must have been the order and not the interactions of the stubbed methods (or maybe it's both, and here I'm actually stubbing the stubbed version - not sure)